loky/backend/context.py

```diff
@@ -146,11 +160,87 @@ def cpu_count():
+
+    cpu_count_user = min(cpu_count_affinity, cpu_count_cfs, cpu_count_loky)
+    aggregate_cpu_count = min(cpu_count_mp, cpu_count_user)
+
+    if only_physical_cores:
+        cpu_count_physical, error_message = _count_physical_cores()
+        if cpu_count_user < cpu_count_mp:
+            # Respect user setting
+            cpu_count = max(cpu_count_user, 1)
+        elif cpu_count_physical == "not found":
+            # Fallback to default behavior
+            if error_message:
+                # warns only the first time
+                warnings.warn(
+                    "Could not find the number of physical cores. "
+                    "Returning the number of logical cores instead.\n"
+                    "You can silence this warning by setting LOKY_CPU_COUNT"
+                    "to the number of cores you want to use.\n"
+                    + error_message)
+            cpu_count = max(aggregate_cpu_count, 1)
+        else:
+            return cpu_count_physical
+    else:
+        cpu_count = max(aggregate_cpu_count, 1)
+
+    return cpu_count
+
+
+def _count_physical_cores():
+    """Return the number of physical cores
+
+    Return None if not found.
+    The value is cached to avoid repeating subprocess calls.
+    """
+    error_message = ""
+
+    # First check if the value is cached
+    global physical_cores_cache
+    if physical_cores_cache is not None:
+        return physical_cores_cache, error_message
+
+    # Not cached yet
+    try:
+        if sys.platform == "linux":
+            cpu_info = subprocess.run(
+                "lscpu --parse=core".split(" "), capture_output=True)
+            cpu_info = cpu_info.stdout.decode("utf-8").splitlines()
+            cpu_info = {line for line in cpu_info if not line.startswith("#")}
+            cpu_count_physical = len(cpu_info)
+        elif sys.platform == "win32":
+            cpu_info = subprocess.run(
+                "wmic CPU Get NumberOfCores /Format:csv".split(" "),
+                capture_output=True)
+            cpu_info = cpu_info.stdout.decode('utf-8').splitlines()
+            cpu_info = [l.split(",")[1] for l in cpu_info
+                        if (l and l != "Node,NumberOfCores")]
+            cpu_count_physical = sum(map(int, cpu_info))
+        elif sys.platform == "darwin":
+            cpu_info = subprocess.run(
+                "sysctl -n hw.physicalcpu".split(" "), capture_output=True)
+            cpu_info = cpu_info.stdout.decode('utf-8')
+            cpu_count_physical = int(cpu_info)
+        else:
+            cpu_count_physical = "not found"
+            error_message = "unsupported platform"
+
+        # if cpu_count_physical < 1, we did not find a valid value
+        if cpu_count_physical != "not found" and cpu_count_physical < 1:
+            cpu_count_physical = "not found"
+            error_message = "found a number of physical cores < 1"
+        
+    except Exception as e:
+        error_message = str(e)
```

Comment:
Maybe we should also record the traceback in the `error_message`.

Reply:
I only did it for python 3. It's a bit tricky to do it in python 2. But I don't think it's necessary because you'll drop python 2 support soon, right ? :D

Reply:
Let's not worry about Python 2 anymore.